jwql/utils/utils.py

```diff
@@ -33,13 +33,108 @@
 import json
 import os
 import re
+import shutil
 
 from jwql.utils import permissions
 from jwql.utils.constants import FILE_SUFFIX_TYPES, JWST_INSTRUMENT_NAMES_SHORTHAND
 
 __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
 
 
+def copy_files(files, out_dir):
+    """Copy a given file to a given directory. Only try to copy the file
+    if it is not already present in the output directory.
+
+    Parameters
+    ----------
+    files : list
+        List of files to be copied
+
+    out_dir : str
+        Destination directory
+
+    Returns
+    -------
+    success : list
+        Files successfully copied (or that already existed in out_dir)
+
+    failed : list
+        Files that were not copied
+    """
+    # Copy files if they do not already exist
+    success = []
+    failed = []
+    for input_file in files:
+        input_new_path = os.path.join(out_dir, os.path.basename(input_file))
+        if os.path.isfile(input_new_path):
+            success.append(input_new_path)
+        else:
+            try:
+                shutil.copy2(input_file, out_dir)
+                success.append(input_new_path)
+            except:
+                failed.append(input_file)
+    return success, failed
```

Comment:
@bhilbert4 It would be handy here to also add a call to `set_permissions` on newly-copied files.

Reply:
done